File: 0205/Code/STUDY_FILES/study12/cli_node/src/words.js
import fs from 'fs';

const FILE_PATH = './data/memo.json';

const getData = () => {
  const f = fs.readFileSync(FILE_PATH, 'utf-8');
  return JSON.parse(f)
}

const setData = (data) => {
  fs.writeFileSync(FILE_PATH, JSON.stringify(data), 'utf-8');
  list();
}

export const list = () => {
  const data = getData();
  if(data.words.length > 0) {
    let line1 = "";
    let line2 = "";
    for(let i = 0; i < 50; i++) {
      line1 += "=";
      line2 += "-";
    }
    console.log(line1);
    console.log(`번호\t내용`);
    data.words.forEach((v, i) => {
      if(i < data.words.length) console.log(line2);
      console.log(`${v.id}\t${v.word}`);
    });
    console.log(line1);
  } else {
    console.log("데이터가 없습니다.")
  }
}

export const insert = (word) => {
  const data = getData();
  const words = data.words;
  const id = (words.length === 0) ? 1 : words.at(-1).id + 1;
  const row = {id, word};
  words.push(row);
  data.words = words;
  setData(data);
}

export const update = (id, word) => {
  const data = getData();
  const words = data.words;
  data.words = words.filter(v => {
    if(v.id === Number(id)) v.word = word;
    return v;
  });
  setData(data);
}

export const remove = (id) => {
  const data = getData();
  const words = data.words;
  data.words = words.filter(v => v.id !== Number(id));
  setData(data);
}

File: 0205/Code/STUDY_FILES/study12/cli_python/src/words.py
import json

FILE_PATH = "./data/memo.json"

def getData():
  f = open(FILE_PATH, "r", encoding="utf-8")
  result = json.load(f)
  f.close()
  return result

def setData(data):
  f = open(FILE_PATH, "w", encoding="utf-8")
  json.dump(data, f, ensure_ascii=False)
  f.close()
  list()

def list():
  data = getData()
  if len(data["words"]) > 0:
    line1 = "=" * 50
    line2 = "-" * 50
    print(line1)
    print(f"번호\t내용")
    for i in range(len(data["words"])):
      if i < len(data["words"]): print(line2)
      print(f"{data["words"][i]["id"]}\t{data["words"][i]["word"]}")
    print(line1)
  else:
    print("데이터가 없습니다.")

def insert(word):
  data = getData()
  id = (max((word["id"] for word in data["words"]), default=0) + 1)
  row = {"id": id, "word": word}
  data["words"].append(row)
  setData(data)

def update(id, word):
  data = getData()
  for i in range(len(data["words"])):
    if data["words"][i]["id"] == int(id):
      data["words"][i]["word"] = word
      break
  setData(data)

def remove(id):
  data = getData()
  for i in range(len(data["words"])):
    if data["words"][i]["id"] == int(id):
      del data["words"][i]
      break
  setData(data)
